core/domain/user_domain.py

```diff
@@ -483,6 +492,10 @@ def unmark_banned(self, default_role: str) -> None:
         self.banned = False
         self.roles = [default_role]
 
+    def set_lesson_info_once_to_true(self) -> None:
```

Comment:
We are not really setting lesson info right, should be something like `set_viewed_lesson_info_once_to_true` .

Reply:
Done!